### PR TITLE
Refactor: Add compare atom symbol between orbital file and pseudo *.upf file 

### PR DESCRIPTION
**Pull request by Zhuxuegang2022**
In order to solve the issue of #2018 .  Initially, a map type for the elements of the periodic table was incorporated into odule_base/atom_in.h. Subsequently, a comparison function was introduced in module_cell/unit_cell.cpp to identify the element type between the orbital file and pseudo *.upf file.

### CODE REVIEW

**Comment by Zhuxuegang2022** [comment]
this issue is to sove the #2018

File: source/module_cell/test/atom_spec_test.cpp [hunk 1]
```diff
@@ -67,7 +68,8 @@ TEST_F(AtomSpecTest, PrintAtom)
 	ofs.close();
 	ifs.open("tmp_atom_info");
 	std::string str((std::istreambuf_iterator<char>(ifs)),std::istreambuf_iterator<char>());
-    	EXPECT_THAT(str, testing::HasSubstr("label = C"));
+	   	EXPECT_THAT(str, testing::HasSubstr("label = C"));
+		EXPECT_THAT(str, testing::HasSubstr("label_orb = C"));
```
Comment (Zhuxuegang2022) [inline source/module_cell/test/atom_spec_test.cpp:72]:
```suggestion
```

File: source/module_cell/test/atom_spec_test.cpp [hunk 2]
```diff
@@ -48,6 +48,7 @@ TEST_F(AtomSpecTest, PrintAtom)
 #endif
 	ofs.open("tmp_atom_info");
 	atom.label = "C";
+	atom.label_orb = "C";
```
Comment (Zhuxuegang2022) [inline source/module_cell/test/atom_spec_test.cpp:51]:
```suggestion
```

File: source/module_cell/test/atom_spec_test.cpp [hunk 3]
```diff
@@ -132,6 +134,7 @@ TEST_F(AtomSpecTest, BcastAtom)
 	if(GlobalV::MY_RANK==0)
 	{
 		atom.label = "C";
+		atom.label_orb = "C";
```
Comment (Zhuxuegang2022) [inline source/module_cell/test/atom_spec_test.cpp:137]:
```suggestion
```

File: source/module_cell/test/atom_spec_test.cpp [hunk 4]
```diff
@@ -172,6 +175,7 @@ TEST_F(AtomSpecTest, BcastAtom)
 	if(GlobalV::MY_RANK!=0)
 	{
 		EXPECT_EQ(atom.label,"C");
+		EXPECT_EQ(atom.label_orb,"C");
```
Comment (Zhuxuegang2022) [inline source/module_cell/test/atom_spec_test.cpp:178]:
```suggestion
```

File: source/module_cell/test/unitcell_test.cpp [hunk 5]
```diff
@@ -278,6 +278,24 @@ TEST_F(UcellDeathTest,SetupWarningQuit2)
...
+{
+    std::string stru_label[] =   {"Ag",  "Ag",  "Ag",     "47", "47", "47",     "Silver", "Silver", "Silver", "Ag",  "Ag", "Ag",        "Ag_empty"};
+    std::string pseudo_label[] = {"Ag",  "47",  "Silver", "Ag", "47", "Silver", "Ag",     "47",     "Silver", "Ag1", "ag", "ag_locpsp", "Ag"      };
+	for (int it = 0; it < 12; it++)
+	{
+	ucell->compare_atom_labels(stru_label[it], pseudo_label[it]);
+	}
+	stru_label[0] = "Fe";
+	pseudo_label[0] = "O";
+	std::string atom_label_in_STRU = "atom label in STRU is ";
```
Comment (Zhuxuegang2022) [inline source/module_cell/test/unitcell_test.cpp:291]:
```suggestion
std::string atom_label_in_STRU = "atom label in orbital file ";
```

File: source/module_cell/unitcell.cpp [hunk 6]
```diff
@@ -1580,4 +1590,61 @@ void UnitCell::cal_nelec(double& nelec)
         ModuleBase::GlobalFunc::OUT(GlobalV::ofs_running, "AUTOSET number of electrons: ", nelec);
     }
     return;
+}
+
+void UnitCell::compare_atom_labels(std::string label1, std::string label2)
+{
+    if (label1 != label2) //'!( "Ag" == "Ag" || "47" == "47" || "Silver" == Silver" )'
+    {	atom_in ai;
```
Comment (hongriTianqi) [inline source/module_cell/unitcell.cpp]:
```suggestion
    {	
        atom_in ai;
```

File: source/module_cell/unitcell.cpp [hunk 7]
```diff
@@ -1580,4 +1590,61 @@ void UnitCell::cal_nelec(double& nelec)
...
+			      std::to_string(ai.atom_Z[stru_label]) == psuedo_label ||   // ' !("Ag1" == "47" )'
+			      ai.atom_symbol[stru_label] == psuedo_label ||              // ' !("Ag1" == "Silver")'
+			      stru_label == std::to_string(ai.atom_Z[psuedo_label]) ||  // ' !("47" == "Ag1" )'
+		          stru_label == std::to_string(ai.symbol_Z[psuedo_label]) || // ' !("47" == "Silver1" )'
+			      stru_label == ai.atom_symbol[psuedo_label] ||              // ' !("Silver1" == "Ag" )'
+			      std::to_string(ai.symbol_Z[stru_label]) == psuedo_label )) // ' !("Silver1" == "47" )'
+            
+				
+			{	
+				std::string atom_label_in_STRU = "atom label in STRU is ";
```
Comment (Zhuxuegang2022) [inline source/module_cell/unitcell.cpp:1644]:
```suggestion
    std::string atom_label_in_orbital = "atom label in orbital file ";
```

File: source/module_cell/unitcell.cpp [hunk 8]
```diff
@@ -1580,4 +1590,61 @@ void UnitCell::cal_nelec(double& nelec)
...
+				std::string mismatch_with_pseudo = " mismatch with pseudo file of ";
+                ModuleBase::WARNING_QUIT("UnitCell::read_pseudo", atom_label_in_STRU + label1 + mismatch_with_pseudo +label2);
```
Comment (Zhuxuegang2022) [inline source/module_cell/unitcell.cpp:1646]:
```suggestion
ModuleBase::WARNING_QUIT("UnitCell::read_pseudo", atom_label_in_orbital + label1 + mismatch_with_pseudo +label2);
```

File: source/module_cell/test/unitcell_test.cpp [hunk 9]
```diff
@@ -278,6 +278,24 @@ TEST_F(UcellDeathTest,SetupWarningQuit2)
...
+	std::string mismatch_with_pseudo = " mismatch with pseudo file of ";
+    testing::internal::CaptureStdout();
+    EXPECT_EXIT(ucell->compare_atom_labels(stru_label[0], pseudo_label[0]),::testing::ExitedWithCode(0),"");
+    output = testing::internal::GetCapturedStdout();
+    EXPECT_THAT(output, testing::HasSubstr(atom_label_in_STRU + stru_label[0] + mismatch_with_pseudo +pseudo_label[0]));
```
Comment (Zhuxuegang2022) [inline source/module_cell/test/unitcell_test.cpp:296]:
```suggestion
EXPECT_THAT(output, testing::HasSubstr(atom_label_in_STRU + stru_label[0] + mismatch_with_pseudo +pseudo_label[0]));
```

File: source/module_cell/unitcell.cpp [hunk 10]
```diff
@@ -671,73 +672,82 @@ void UnitCell::read_pseudo(std::ofstream &ofs)
...
-                        << " " << atom->ncpp.dion(ib,ib2)<<std::endl;
-                }
-            }
-            ofs << "</DIJ>" << std::endl;
-
-            for(int i=0; i<atom->ncpp.nbeta; i++)
+		for (int it = 0; it < this->ntype; it++)
+    	{
+    	    Atom* atom = &atoms[it];
+    	    compare_atom_labels(atom->label_orb, atom->ncpp.psd);
```
Comment (Zhuxuegang2022) [inline source/module_cell/unitcell.cpp:680]:
```suggestion
if (!(atom->label_orb.empty()))
{
compare_atom_labels(atom->label_orb, atom->ncpp.psd);
}
```

File: source/module_cell/unitcell.cpp [hunk 11]
```diff
@@ -1580,4 +1593,62 @@ void UnitCell::cal_nelec(double& nelec)
         ModuleBase::GlobalFunc::OUT(GlobalV::ofs_running, "AUTOSET number of electrons: ", nelec);
     }
     return;
+}
+
+void UnitCell::compare_atom_labels(std::string label1, std::string label2)
+{
+    if (label1 != label2) //'!( "Ag" == "Ag" || "47" == "47" || "Silver" == Silver" )'
+    {	
+		atom_in ai;
```
Comment (hongriTianqi) [inline source/module_cell/unitcell.cpp]:
```suggestion
        atom_in ai;
```

**Review by hongriTianqi — APPROVED** [review]

**Pull request merged**